src/plugins/intel_cpu/src/nodes/mha.cpp

```diff
@@ -0,0 +1,1086 @@
+            add(reg_in, jcp_.src_prc.size() * jcp_.src_stride);
+            add(reg_out, jcp_.dst_prc.size() * jcp_.dst_stride);
+
+            jmp(convert_reorder_outter_loop_label, T_NEAR);
+        }
+        L(convert_reorder_outter_end_label);
+
+        this->postamble();
+
+        load_emitter->emit_data();
+        store_emitter->emit_data();
+    }
+
+    void convert_reorder(size_t step) {
+        bool is_tail = step < vec_size;
+
+        load(vmm_in, reg_in_aux, jcp_.src_prc, step, is_tail);
+
+        if (jcp_.src_prc == Precision::I32) {
+            if (jcp_.with_scales) {
+                if (!jcp_.broadcast_scales) {
+                    load(vmm_scales, reg_scales, Precision::FP32, step, is_tail);
+                    add(reg_scales,  sizeof(float) * step);
+                }
+                uni_vmulps(vmm_in, vmm_in, vmm_scales);
+            }
+        }
+
+        store(reg_out_aux, vmm_in, jcp_.dst_prc, step);
+
+        if (!is_tail) {
+            add(reg_in_aux, jcp_.src_prc.size() * step);
+            add(reg_out_aux, jcp_.dst_prc.size() * step);
+        }
+    }
+#undef GET_OFF
+
+    inline void load(const Vmm& vmm_dst, const Xbyak::Reg64& reg_src, Precision src_prc, const int& elt_num, bool fill) {
+        load_emitter->emit_code({static_cast<size_t>(reg_src.getIdx())}, {static_cast<size_t>(vmm_dst.getIdx())},
+            std::make_shared<load_emitter_context>(src_prc, Precision::FP32, elt_num, 0, fill, "float_min"),
+            pool_aux_vmm_idxs, pool_aux_gpr_idxs);
+    }
+    inline void store(const Xbyak::Reg64& reg_dst, const Vmm& vmm_src, Precision dst_prc, const int& elt_num) {
+        store_emitter->emit_code({static_cast<size_t>(vmm_src.getIdx())}, {static_cast<size_t>(reg_dst.getIdx())},
+            std::make_shared<store_emitter_context>(Precision::FP32, dst_prc, elt_num, 0),
+            pool_aux_vmm_idxs, pool_aux_gpr_idxs);
+    }
+
+    size_t vec_size;
+
+    Xmm xmm_tmp = Xmm(0);
+    Vmm vmm_scales = Vmm(0);
+    Vmm vmm_in = Vmm(1);
+
+    Reg64 reg_in = r8;
+    Reg64 reg_in_aux = r9;
+    Reg64 reg_out = r10;
+    Reg64 reg_out_aux = r11;
+    Reg64 reg_scales = r12;
+    Reg64 reg_inner_work_amount = r14;
+    Reg64 reg_outter_work_amount = r15;
+    Reg64 reg_params = abi_param1;
+
+    const std::vector<size_t> pool_aux_gpr_idxs = { static_cast<size_t>(rsi.getIdx()), static_cast<size_t>(rbp.getIdx()) };
+    const std::vector<size_t> pool_aux_vmm_idxs = { static_cast<size_t>(xmm_tmp.getIdx()) };
+
+    std::unique_ptr<jit_load_emitter> load_emitter = nullptr;
+    std::unique_ptr<jit_store_emitter> store_emitter = nullptr;
+};
+
+
+bool MHA::isSupportedOperation(const std::shared_ptr<const ov::Node>& op, std::string& errorMessage) noexcept {
+    try {
+        const auto mha = std::dynamic_pointer_cast<const MHANode>(op);
+        if (!mha) {
+            errorMessage = "Only MHA from CPU internal opset is supported";
+            return false;
+        }
+
```

Comment:
Could we add condition for `dnnl::impl::cpu::x64::avx512_core_vnni` here?

Reply:
Good catch! Added condition for supported precisions and isa.